src/main/java/edu/kis/powp/jobs2d/command/ExceedingCanvasCommandVisitor.java

```diff
@@ -0,0 +1,46 @@
+package edu.kis.powp.jobs2d.command;
+
+import edu.kis.powp.jobs2d.command.gui.Canvas;
+import java.awt.Point;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class ExceedingCanvasCommandVisitor implements CommandVisitor {
+
+    private Canvas canvas;
+    private boolean isExceedingCanvas = false;
+    private List<Point> commands = new ArrayList<>();
+
+    public ExceedingCanvasCommandVisitor(Canvas canvas) {
+        this.canvas = canvas;
+    }
+
+
+    @Override
+    public void visit(OperateToCommand operateToCommand) {
+        canvas.isPointWithinBounds(operateToCommand.getX(), operateToCommand.getY());
+        isExceedingCanvas |= !canvas.isPointWithinBounds(operateToCommand.getX(), operateToCommand.getY());
+    }
+
+    @Override
+    public void visit(SetPositionCommand setPositionCommand) {
+        isExceedingCanvas |= !canvas.isPointWithinBounds(setPositionCommand.getX(), setPositionCommand.getY());
+    }
```

Comment:
OK, but this approach is not stateless. Visitor is not reusable, Running visitation again on a 'proper' command would not change the result.